floris/tools/uncertainty_interface.py

```diff
@@ -1,337 +1,103 @@
+from __future__ import annotations
 
-import copy
+from pathlib import Path
 
 import numpy as np
-from scipy.stats import norm
 
-from floris.logging_manager import LoggingManager
 from floris.tools import FlorisInterface
-from floris.utilities import wrap_360
+from floris.tools.wind_data import WindDataBase
+from floris.type_dec import (
+    floris_array_converter,
+    NDArrayBool,
+    NDArrayFloat,
+)
 
 
-class UncertaintyInterface(LoggingManager):
-    def __init__(
-        self,
-        configuration,
-        unc_options=None,
-        unc_pmfs=None,
-        fix_yaw_in_relative_frame=False,
-    ):
-        """A wrapper around the nominal floris_interface class that adds
-        uncertainty to the floris evaluations. One can specify a probability
-        distribution function (pdf) for the ambient wind direction. Unless
-        the exact pdf is specified manually using the option 'unc_pmfs', a
-        Gaussian probability distribution function will be assumed.
+class UncertaintyInterface(FlorisInterface):
```

Comment:
Reading through this class, I wonder if inheritance is the right architecture over composition. Instead of managing the FlorisInterface through `self` you could create it in `__init__` here and call `.set()` as needed. This avoids raising NotImplementErrors at the end and doesn't necessarily tie this class to the other. Note that the two functions used from the super-class are `.set` and `.get_turbine_power`, and they're both overloaded. This is just to say that the benefit of inheritance might be outweighed by any complexities that come up when theres a change to `FlorisInterface` that shouldn't impact `UncertaintyInterface` (i.e. a new function to do an unrelated task will require adding the NotImplemented-version here if you want to maintain the pattern).

Reply:
This would require adding a `UncertaintyInterface.run()` function, fwiw

Reply:
We're also using the init() function from the super.  @misi9170 and I debated this at the beginning and thought the inheritance was slightly more elegant, but I think I was thinking if we did composition we'd have to pass in the "certain" fi into the uncertain, but I think your solution is really nice, if I understand it right, we declare the "certain" fi within the uncertain one and keep it around as a member to call on as needed.  Do I have that right?  I can code it up,

Reply:
Yea, I think you have it right. Here's a prototype of how this would connect to FlorisInterface with the current functionality:

```python
class UncertaintyInterface:

    def __init__(self, ...):
        self.floris_interface = FlorisInterface(...)

    def set(self, ...):
        self.floris_interface.set(...)
        self.set_uncertain()

    def run(self):
        self.floris_interface.run()

    def get_turbine_power(self):
        unique_turbine_powers = self.floris_interface.get_turbine_power()
        expanded...
        weights...
        return result
```

For what it's worth, I'm not sure if this is "right" either, but it's another option. What stood out is that usually a subclass is a superset of the parent-class and in this case it's a subset. That's more of a smell test than a rule, so I thought it would be worth considering composition.

Reply:
ok! last commit switches from inheritance to composition,